packages/web-api-scan-runner/src/runner/runner.ts

```diff
@@ -0,0 +1,138 @@
+            browser = await this.webDriverTask.launch();
+
+            await this.scan(pageScanResult);
+        } catch (error) {
+            this.logger.logInfo(`Scan failed ${error}`);
+            pageScanResult.run = this.getRunResult('failed', error instanceof Error ? error.message : `${error}`);
+            pageScanResult.scanResult = { state: 'unknown', issueCount: 0 };
+            pageScanResult.reports = [];
+        } finally {
+            try {
+                await this.webDriverTask.close();
+            } catch (error) {
+                this.logger.logError(`unable to close web driver ${error}`);
+            }
+        }
+
+        this.logger.logInfo(`Updating page scan in database`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+    }
+
+    private async scan(pageScanResult: OnDemandPageScanResult): Promise<void> {
+        this.logger.logInfo(`Running Scan`);
+
+        const axeScanResults = await this.scannerTask.scan(pageScanResult.url);
+
+        if (!isNil(axeScanResults.error)) {
+            this.logger.logInfo(`Changing page status to failed`);
+            pageScanResult.run = this.getRunResult('failed', axeScanResults.error);
+            pageScanResult.scanResult = { state: 'unknown', issueCount: 0 };
+            pageScanResult.reports = [];
+        } else {
+            this.logger.logInfo(`Changing page status to completed`);
+            pageScanResult.run = this.getRunResult('completed');
+            pageScanResult.scanResult = this.getScanStatus(axeScanResults);
+            pageScanResult.reports = [await this.saveScanReport(axeScanResults)];
+        }
+    }
+
+    private getRunResult(state: OnDemandPageScanRunState, error?: string): OnDemandPageScanRunResult {
+        return {
+            state: state,
+            timestamp: new Date().toJSON(),
+            error,
+        };
+    }
+
+    private getScanStatus(axeResults: AxeScanResults): OnDemandScanResult {
+        let issueCount = 0;
+        let state: ScanState = 'fail';
+
+        if (axeResults.results.violations !== undefined && axeResults.results.violations.length > 0) {
+            issueCount = axeResults.results.violations.length;
+            state = 'fail';
+        } else {
+            state = 'pass';
+            issueCount = 0;
+        }
+
+        return {
+            state,
+            issueCount,
+        };
+    }
+
+    private async saveScanReport(axeResults: AxeScanResults): Promise<OnDemandPageScanReport> {
+        const reportId = this.guidGenerator.createGuid();
+        let href: string;
+        const format = 'sarif';
+
+        this.logger.logInfo(`Converting to Sarif...`);
+        const sarifResults: SarifLog = this.convertAxeToSarifFunc(axeResults.results);
+
+        this.logger.logInfo(`Saving sarif results to Blobs...`);
+        await this.pageScanRunReportService.saveSarifReport(reportId, JSON.stringify(sarifResults));
```

Comment:
The provide can be refactored for saveSarifReport() API, it is logically to return the blob path instead of getting it from separate call

Reply:
out of scope, could be done in another PR since it's non blocking issue